plugins/metrics/message.go

```diff
@@ -70,9 +106,33 @@ var (
 	// current number of missing messages in missingMessageStorage
 	missingMessageCountDB atomic.Uint64
 
+	// current number of finalized messages
+	finalizedMessageCount = make(map[MessageType]uint64)
+
+	// protect map from concurrent read/write.
+	finalizedMessageCountMutex syncutils.RWMutex
+
+	// total time it took all messages to finalize. unit is milliseconds!
```

Comment:
we choose to go for ns everywhere
to be more precise we should use time.Duration where possible when discussing time elapsed or intervals, and use uint64 in nanoseconds where not possible